Add docs for `anchors` key (#7726)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/core-upgrade/05-upgrading-to-v1.10.md

Large data sets can slow down dbt build times, making it harder for developers to test new code efficiently. The [`--sample` flag](/docs/build/sample-flag), available for the `run` and `build` commands, helps reduce build times and warehouse costs by running dbt in sample mode. It generates filtered refs and sources using time-based sampling, allowing developers to validate outputs without building entire models.

> ### New `anchors:` key

Author: ```suggestion
### Move standalone anchors under `anchors:` key
```

Large data sets can slow down dbt build times, making it harder for developers to test new code efficiently. The [`--sample` flag](/docs/build/sample-flag), available for the `run` and `build` commands, helps reduce build times and warehouse costs by running dbt in sample mode. It generates filtered refs and sources using time-based sampling, allowing developers to validate outputs without building entire models.

### New `anchors:` key

> You can use the new `anchors:` key to reuse configuration blocks across your dbt project files. Note that dbt Core v1.10 does not support standalone anchor definitions at the top level of YAML files. Standalone anchors that are meant to be used as reusable snippets should be moved under the `anchors:` key. For example:

Reviewer: ```suggestion
As part of the ongoing process of making the dbt authoring language more precise, dbt Core v1.10 will emit a warning when it sees a standalone anchor definition at the top level of a YAML file. Instead, you can use the new top-level `anchors:` key as a container for those reusable configuration blocks. For example:
```

Addressing two issues here: 
- dbt Core 1.10 will not break functionality present in 1.9, so "does not support" is inaccurate
- It is not possible to use the `anchors` key to reuse configuration across multiple files, only inside of a single file

<File name='models/_models.yml'>

```yml
anchors:
  - columns: &id_column
    - name: id
      description: This is a unique identifier.
      data_tests:
        - not_null

models:
  - name: my_first_model
    columns: *id_column
  - name: my_second_model
    columns: *id_column
```

</File>

> Note that not all anchors should be moved under an `anchors:` block, such as anchors that are part of the main YAML structure (for example, defining tests on a column).

Reviewer: I would like to see a more fully-fleshed out YAML file here (and in the Fusion upgrade guide) which contains anchors interspersed with other hardcoded columns in the models section.

Reviewer 2: Thanks @joellabes Can you point us towards something we can crib from to get this example? Any environments you're aware of with a fleshed-out implementation?

Reviewer: Ugh this was a note-to-self to come back and provide one. I don't have anything super fleshed out, but while experimenting yesterday I made this. Note that this example has a single element as its anchored value, not an array, which makes it possible to specify more columns. 

Note also that the key name in the anchor is irrelevant - discovering this was a big mental unlock for me, because most examples have the name of the key matching the place where the anchor is invoked later, and I didn't understand how that worked. Turns out it doesn't! 

```
# Before
this_value_is_always_ignored: &the_anchor_name_is_what_matters
  name: id
  description: This is a unique identifier.
  data_tests:
    - not_null

# After

anchors:
  - this_value_is_always_ignored: &the_anchor_name_is_what_matters
      name: id
      description: This is a unique identifier.
      data_tests:
        - not_null

models:
    - name: my_first_model
      description: This is a description for the first model
      columns:
        - *the_anchor_name_is_what_matters
        - name: my_second_column
          description: "This is a description for the second column. I can add it even though there's another column added via anchor"
          data_tests:
            - not_null

    - name: my_second_model
      description: This is a description for the second model
      columns:
        - *the_anchor_name_is_what_matters

```

> Note that not all anchors should be moved under an `anchors` block. Some anchors are part of the main YAML structure (for example, defining tests on a column) and should _not_ be moved under a top-level `anchors` key.

Reviewer 2: Small suggestion to tighten this up a little
```suggestion
Note that not all anchors should be moved under an `anchors:` block, such as anchors that are part of the main YAML structure (for example, defining tests on a column).
```

Large data sets can slow down dbt build times, making it harder for developers to test new code efficiently. The [`--sample` flag](/docs/build/sample-flag), available for the `run` and `build` commands, helps reduce build times and warehouse costs by running dbt in sample mode. It generates filtered refs and sources using time-based sampling, allowing developers to validate outputs without building entire models.

> ### New `anchors:` key

Reviewer 3: Whatever title you use for Fusion you could also use here.
